src/models/state/wallet/wallet_state.rs

```diff
@@ -286,6 +285,29 @@ impl WalletState {
             .collect_vec()
     }
 
+    // returns true if the utxo can be unlocked by one of the
+    // known wallet keys.
+    pub fn is_wallet_utxo(&self, utxo: &Utxo) -> bool {
+        self.get_known_spending_keys()
+            .iter()
+            .map(|k| k.to_address().lock_script().hash())
+            .any(|h| h == utxo.lock_script_hash)
+    }
+
+    // TODO: These spending keys should probably be derived dynamically from some
+    // state in the wallet. And we should allow for other types than just generation
+    // addresses.
+    //
+    // Probably the wallet should keep track of index of latest derived key
+    // that has been requested by the user for purpose of receiving
+    // funds.  We could also perform a sequential scan at startup (or import)
+    // of keys that have received funds, up to some "gap".  In bitcoin/bip32
+    // this gap is defined as 20 keys in a row that have never received funds.
```

Comment:
Performing a sequential scan at startup or import is a fantastic idea. I think 20 is a fine default value for the gap but this value should be overridable through command line arguments.